modules/core/src/lib/layer.ts

```diff
@@ -332,13 +333,21 @@ export default abstract class Layer<PropsT extends {} = {}> extends Component<
     return (state && (state.models || (state.model && [state.model]))) || [];
   }
 
+  // TODO deprecate in favour of setShaderInputs
   /** Update shader module parameters */
   setModuleParameters(moduleParameters: any): void {
     for (const model of this.getModels()) {
       model.updateModuleSettings(moduleParameters);
     }
   }
 
+  /** Update shader input parameters */
+  setShaderInputs(...props: Parameters<Model['shaderInputs']['setProps']>): void {
```

Comment:
Good for now but we should probably audit the API... Perhaps `setShaderModuleProps()` would be more correct?